tests/MongoDB.Driver.Tests/MongoCollectionProseTests.cs

```diff
@@ -0,0 +1,83 @@
+using MongoDB.Driver.Core.Bindings;
+using MongoDB.Driver.Core.Clusters.ServerSelectors;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.TestHelpers;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using Xunit;
+
+namespace MongoDB.Driver.Tests
+{
+    public class MongoCollectionProseTests
+    {
+        [SkippableFact]
+        public void WriteConcernError_errInfo_should_be_propagated()
+        {
+            RequireServer.Check().Supports(Feature.FailPointsFailCommand);
+
+            var failPointCommand = @"
+                {
+                    ""configureFailPoint"" : ""failCommand"",
+                    ""data"" : {
+                        ""failCommands"" : [""insert""],
+                        ""writeConcernError"" : {
+                            ""code"" : 100,
+                            ""codeName"" : ""UnsatisfiableWriteConcern"",
+                            ""errmsg"" : ""Not enough data-bearing nodes"",
+                            ""errInfo"" : {
+                                ""writeConcern"" : {
+                                    ""w"" : 2,
+                                    ""wtimeout"" : 0,
+                                    ""provenance"" : ""clientSupplied""
+                                }
+                            }
+                        }
+                    },
+                    ""mode"": { ""times"": 1 }
+                }";
+
+            using (ConfigureFailPoint(failPointCommand))
+            {
+                var client = DriverTestConfiguration.Client;
+                var database = client.GetDatabase(DriverTestConfiguration.DatabaseNamespace.DatabaseName);
+                var collection = database.GetCollection<BsonDocument>(DriverTestConfiguration.CollectionNamespace.CollectionName);
+
+                var exception = Record.Exception(() => collection.InsertOne(new BsonDocument()));
+
+                exception.Should().NotBeNull();
+                var e = exception.InnerException.Should().BeOfType<MongoBulkWriteException<BsonDocument>>().Subject;
+                var writeConcernError = e.WriteConcernError;
+                writeConcernError.Code.Should().Be(100);
+                writeConcernError.CodeName.Should().Be("UnsatisfiableWriteConcern");
+                writeConcernError.Details.Should().Be("{ \"writeConcern\" : { \"w\" : 2, \"wtimeout\" : 0, \"provenance\" : \"clientSupplied\" } }");
+                writeConcernError.Message.Should().Be("Not enough data-bearing nodes");
+            }
+        }
+
+        // private methods
+        private FailPoint ConfigureFailPoint(string failpointCommand)
+        {
+            var cluster = DriverTestConfiguration.Client.Cluster;
+            var server = cluster.SelectServer(WritableServerSelector.Instance, CancellationToken.None);
+            var session = NoCoreSession.NewHandle();
+            return FailPoint.Configure(cluster, session, BsonDocument.Parse(failpointCommand));
```

Comment:
One other minor comment, you can pass `server` instead of `cluster` to `FailPoint.Configure`:

    return FailPoint.Configure(server, session, BsonDocument.Parse(failpointCommand));

Reply:
Or alternatively, remove the `server` variable?